### PR TITLE
Improve oidc discovery integration tests

### PR DESCRIPTION
**Pull request by Thumimku**
Public Issue : https://github.com/wso2/product-is/issues/21313

Current integration suite for oidc discovery has following issues.
- Factory method aims to run 4 scenarios for web finger but eventually it runs oidc discovery test case for super tenant for time unnecessary.
- Current test suite doesn't test sub tenant invalid tenant.
- Current test suite doesn't validate popular fine grand values of the response.
- Current test suite doesn't validate for invalid issuer.

This PR fix the above mentioned issues along with revamping the test class.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCDiscoveryTestCase.java [hunk 1]
```diff
@@ -30,57 +33,59 @@
...
 
     }
 
-    @Test(alwaysRun = true, groups = "wso2.is", description = "webfinger test")
-    public void testWebFinger() throws IOException {
+    @Test(alwaysRun = true, groups = "wso2.is", description = "webfinger test",
+            dataProvider = "webfingerConfigProvider")
+    public void testWebFinger(DiscoveryConfig config) {
 
+        String relUri = "http://openid.net/specs/connect/1.0/issuer";
```
Comment (asha15) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCDiscoveryTestCase.java:78]: Shall we give a meaningful name?

Comment (madurangasiriwardena) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCDiscoveryTestCase.java:78]: I think this is meaningful enough. The value is called `rel`

Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCDiscoveryTestCase.java:78]: relURI -> relationship url.

The rel parameter allows clients to specify what type of information they are seeking about the subject. here we are looking for an OpenID Connect issuer for the given subject.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCDiscoveryTestCase.java [hunk 2]
```diff
@@ -100,21 +106,47 @@ public void testDiscovery() throws IOException {
...
+                {"", "oidcdiscovery"},
+                {"carbon.super", "oidcdiscovery"},
+                {"wso2.com", "oidcdiscovery"},
+                {"", "token"},
+                {"carbon.super", "token"},
+                {"wso2.com", "token"},
+        };
+    }
 
     @DataProvider(name = "webfingerConfigProvider")
```
Comment (asha15) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCDiscoveryTestCase.java:147]:
```suggestion
    @DataProvider(name = "webFingerConfigProvider")
```

**Review by madurangasiriwardena — APPROVED** [review]

**Pull request merged**